src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/DiscoverServiceController.java

```diff
@@ -23,37 +23,38 @@
 @RestController
 @RequestMapping("/v3")
 @Validated
-public class DataSourceController {
+public class DiscoverServiceController {
 
-  @Autowired private DataSourceService dataSourceService;
+  @Autowired private DiscoveryServiceService discoveryServiceService;
   @Autowired private ModelMapper modelMapper;
 
-  @GetMapping("/data-sources")
-  List<DataSourceDTO> list() {
-    return dataSourceService.findAll();
+  @GetMapping("/discovery-service")
+  List<DiscoveryServiceDTO> list() {
+    return discoveryServiceService.findAll();
   }
 
-  @GetMapping("/data-sources/{id}")
-  DataSourceDTO retrieve(@PathVariable Long id) {
-    return dataSourceService.findById(id);
+  @GetMapping("/discovery-service/{id}")
+  DiscoveryServiceDTO retrieve(@PathVariable Long id) {
+    return discoveryServiceService.findById(id);
   }
 
   @PostMapping(
-      value = "/data-sources",
+      value = "/discovery-service",
       consumes = MediaType.APPLICATION_JSON_VALUE,
       produces = MediaType.APPLICATION_JSON_VALUE)
   @ResponseStatus(HttpStatus.CREATED)
-  DataSourceDTO add(@Validated(Create.class) @RequestBody DataSourceCreateDTO request) {
-    return dataSourceService.create(request);
+  DiscoveryServiceDTO add(@Validated(Create.class) @RequestBody DiscoveryServiceCreateDTO request) {
+    return discoveryServiceService.create(request);
   }
 
   @PutMapping(
-      value = "/data-sources/{id}",
+      value = "/discovery-service/{id}",
       consumes = MediaType.APPLICATION_JSON_VALUE,
       produces = MediaType.APPLICATION_JSON_VALUE)
   @ResponseStatus(HttpStatus.NO_CONTENT)
-  DataSourceDTO update(
-      @PathVariable Long id, @Validated(Update.class) @RequestBody DataSourceCreateDTO request) {
-    return dataSourceService.update(id, request);
+  DiscoveryServiceDTO update(
+      @PathVariable Long id,
+      @Validated(Update.class) @RequestBody DiscoveryServiceCreateDTO request) {
+    return discoveryServiceService.update(id, request);
   }
 }
```

Comment:
Please change to /discovery-services, see https://restfulapi.net/resource-naming/